crates/runtime/src/lib.rs

```diff
@@ -142,6 +145,29 @@ impl<'a, T> GuestPtr<'a, [T]> {
         let base = GuestPtr::new(self.mem, self.offset_base());
         (0..self.len()).map(move |i| base.add(i))
     }
+
+    /// Safety note: The user must use the same `GuestBorrows` to check each
+    /// use of GuestType::as_raw that is alive simultaneously.
+    /// The resulting `*mut [T]` can be used as a `&mut [T]` as long as the
+    /// reference is dropped before any Wasm code is re-entered.
+    pub fn as_raw(&self, bc: &mut GuestBorrows) -> Result<*mut [T], GuestError> {
+        let ptr = self
+            .mem
+            .validate_size_align(self.pointer.0, T::guest_align(), self.pointer.1)?
```

Comment:
I think `self.pointer.1` here wants to be `self.pointer.1.checked_mul(T::guest_size())`

Reply:
Good catch! I think we need a different error variant for expressing when pointer arithmetic overflows, versus is out-of-bounds.